Blog feature

amritmalhi99 opened the pull request.

amritmalhi99 committed:
Create blog.js

Blog feature that allows customers to post blog entries. The blog structure will include different user roles (customers, sellers, experts) and allow only customers to create posts.

amritmalhi99 committed:
Create blog.js

amritmalhi99 committed:
blog.js

amritmalhi99 committed:
Create blogPosts.handlebars

Displays a list of all blog posts.

amritmalhi99 committed:
Create singleBlog.handlebars

Displays the details of a single blog post.

amritmalhi99 committed:
Create createBlog.handlebars

Displays a form for creating a new blog post.

amritmalhi99 committed:
Create editBlog.handlebars

Displays a form for editing an existing blog post.

amritmalhi99 committed:
Create error.handlebars

Displays error messages when something goes wrong

amritmalhi99 committed:
Create mainblog.handlebars

nuendo2004 commented on src/data/blog.js:
Please use database for storing the blogs, not just local array

nuendo2004 replied:
Please use import from es6, not required. And run the code, with UI all of that. Only add pull request if the code runs, please don't assume it will run. Also add a new tab or button for the blogs, I should be able to navigate to the page as a user.
We are not using mongoose, we are using mongodb driver, please reference code from lecture

amritmalhi99 committed:
Update blog.js

This file defines the MongoDB schema for the blog posts and includes custom validators.

amritmalhi99 committed:
Update blog.js

This file contains all the routes for handling blog operations, including viewing all blogs, creating a new blog, editing, and deleting.

amritmalhi99 committed:
Update mainblog.handlebars

This is the main layout that will be extended by other views.

amritmalhi99 committed:
Update blogPosts.handlebars

This view displays a list of all blog posts.

amritmalhi99 committed:
Update singleBlog.handlebars

This view displays the details of a single blog post.

amritmalhi99 committed:
Update createBlog.handlebars

This view displays a form for creating a new blog post.

amritmalhi99 committed:
Update editBlog.handlebars

This view displays a form for editing an existing blog post.

amritmalhi99 committed:
Update error.handlebars

This view displays error messages.

amritmalhi99 committed:
Update blog.js

amritmalhi99 committed:
Update blog.js
